tekton/pipeline_run.go

```diff
@@ -48,24 +49,69 @@ var (
 	ReleaseWorkspaceLabel = fmt.Sprintf("%s/%s", releaseLabelPrefix, "workspace")
 )
 
-// CreateReleasePipelineRun creates a PipelineRun from a given Release and ReleaseStrategy.
-// Release details are added to the labels of the new PipelineRun to be able to reference it later on.
-func CreateReleasePipelineRun(release *v1alpha1.Release, releaseStrategy *v1alpha1.ReleaseStrategy) *tektonv1beta1.PipelineRun {
-	return &tektonv1beta1.PipelineRun{
+// ReleasePipelineRun is a PipelineRun alias, so we can add new methods to it in this file.
+type ReleasePipelineRun struct {
+	tektonv1beta1.PipelineRun
+}
+
+// NewReleasePipelineRun creates an empty PipelineRun in the given namespace. The name will be autogenerated,
+// using the prefix passed as an argument to the function.
+func NewReleasePipelineRun(prefix, namespace string) *ReleasePipelineRun {
+	pipelineRun := tektonv1beta1.PipelineRun{
 		ObjectMeta: v1.ObjectMeta{
-			GenerateName: releaseStrategy.Name + "-",
-			Labels: map[string]string{
-				PipelinesTypeLabel:    PipelineTypeRelease,
-				ReleaseNameLabel:      release.Name,
-				ReleaseWorkspaceLabel: release.Namespace,
-			},
-			Namespace: releaseStrategy.Namespace,
+			GenerateName: prefix + "-",
+			Namespace:    namespace,
 		},
-		Spec: tektonv1beta1.PipelineRunSpec{
-			PipelineRef: &tektonv1beta1.PipelineRef{
-				Name:   releaseStrategy.Spec.Pipeline,
-				Bundle: releaseStrategy.Spec.Bundle,
-			},
+	}
+
+	return &ReleasePipelineRun{pipelineRun}
+}
+
+// AsPipelineRun casts the ReleasePipelineRun to PipelineRun, so it can be used in the Kubernetes client.
+func (r *ReleasePipelineRun) AsPipelineRun() *tektonv1beta1.PipelineRun {
+	return &r.PipelineRun
+}
+
+// WithExtraParam adds an extra param to the release PipelineRun.
```

Comment:
What do you think about adding a comment along the lines of "If the parameter is not part of the invoked release Pipeline definition, it is silently ignored"

I say this just because when I was implementing this, it is something I thought of and tested